DataCollector/DeviceDataCollector.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the MobileDetectBundle.
+ *
+ * (c) Nikolay Ivlev <nikolay.kotovsky@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SunCat\MobileDetectBundle\DataCollector;
+
+use SunCat\MobileDetectBundle\Helper\DeviceView;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+
+/**
+ * DeviceDataCollector class
+ *
+ * @author Jonas HAOUZI <haouzijonas@gmail.com
+ *
+ */
+class DeviceDataCollector extends DataCollector
+{
+    /**
+     * @var DeviceView
+     */
+    protected $deviceView;
+
+    /**
+     * DeviceDataCollector constructor.
+     *
+     * @param DeviceView $deviceView Device View Detector
+     */
+    public function __construct(DeviceView $deviceView)
+    {
+        $this->deviceView = $deviceView;
+    }
+
+    /**
+     * Collects data for the given Request and Response.
+     *
+     * @param Request    $request   A Request instance
+     * @param Response   $response  A Response instance
+     * @param \Exception $exception An Exception instance
+     *
+     * @api
+     */
+    public function collect(
+        Request $request,
+        Response $response,
+        \Exception $exception = null
+    ) {
+        $this->data['currentView'] = $this->deviceView->getViewType();
+        $this->data['views'] = [
+            [
```

Comment:
MobileDetectBundle claims to be compatible with PHP 5.3. Thus, you have to avoid the short array syntax.